js/battle.js

```diff
@@ -4761,6 +4761,7 @@ var Battle = (function () {
 					actions += "" + poke.getName() + " fell straight down!";
 					poke.removeVolatile('magnetrise');
 					poke.removeVolatile('telekinesis');
+					if (poke.lastmove === 'fly' || poke.lastmove === 'bounce') poke.sprite.animReset();
```

Comment:
With "message", I meant the `|-start|smackdown` that the server sends to the client if Smack Down has hit something flying, that would cause the anim to be reset.
If I use Dive with Latios and it gets hit with Smack Down while diving, this message would be sent, because Smack Down has removed Levitate from Latios. If we unconditionally reset the anim then, Latios will not appear to be diving anymore. This does not reflect the state of the game, as Dive is not canceled, whereas Fly/Bounce would be.